Storage/Rust/RustLogins.swift

```diff
@@ -496,4 +713,67 @@ public class RustLogins {
 
         return deferred
     }
+    
+    private func migrateSQLCipherDBIfNeeded() {
+        let keychain = KeychainWrapper.sharedAppContainerKeychain
+        let rustKeys: RustLoginEncryptionKeys = RustLoginEncryptionKeys()
+        let sqlCipherLoginsKey: String? = keychain.string(forKey: rustKeys.loginsUnlockKeychainKey)
+        let sqlCipherLoginsSalt: String? = keychain.string(forKey: rustKeys.loginsSaltKeychainKey)
+        let key = try! self.getStoredKey()
+        
+        // If the sqlcipher salt or key are missing don't migrate
+        if ((sqlCipherLoginsKey ?? "").isEmpty || (sqlCipherLoginsSalt ?? "").isEmpty ) {
+            return
+        }
+        
+        defer {
+            // Delete the old key and salt regardless of if the migration succeeded.  If
+            // it failed, it's just going to fail again next time.
+            keychain.removeObject(forKey: rustKeys.loginsUnlockKeychainKey, withAccessibility: .afterFirstUnlock)
+            keychain.removeObject(forKey: rustKeys.loginsSaltKeychainKey, withAccessibility: .afterFirstUnlock)
+        }
+        let metrics = try! migrateLogins(path: self.perFieldDatabasePath, newEncryptionKey: key, sqlcipherPath: sqlCipherDatabasePath, sqlcipherKey: sqlCipherLoginsKey!, salt: sqlCipherLoginsSalt)
+        // TODO: Report metrics via glean
+    }
+    
+    public func getStoredKey() throws -> String {
+        let rustKeys = RustLoginEncryptionKeys()
+        let key = rustKeys.keychain.string(forKey: rustKeys.loginPerFieldKeychainKey)
+        let encryptedCanaryPhrase = rustKeys.keychain.string(forKey: rustKeys.canaryPhraseKey)
+        
+        // We expected the key to be present, and it is.
+        if key != nil && encryptedCanaryPhrase != nil {
```

Comment:
This is totally optional. But since key and encrypted canary phrase are optionals (no pun intended), swift has nice tuple matching with optionals:
```swift
switch(key1, key2) {
  case (.some(key1), .some(key2)):
  ...
  case (.some(key1), .none):
  case (.none, .some(key2)):
  case (.none, .none):
  ...
```

don't know if it improves indentation much but just throwing it out there for a cool swift thing!